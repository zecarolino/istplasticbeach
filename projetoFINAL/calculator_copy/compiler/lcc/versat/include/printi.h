#ifndef _PRINTI_H_
#define _PRINTI_H_

#include "mod.h"
#include "div.h"
#include "putchar.h"

void printi(int n, int b) {
	int a, c;
	if (a = n / b) printi(a, b);
	c = n % b;
	if (c <= 9) putchar(c + '0');
	else putchar(c + 'A' -10);
}

#endif /* _PRINTI_H_ */
