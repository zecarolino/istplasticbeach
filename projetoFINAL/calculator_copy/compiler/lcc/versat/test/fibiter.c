#include "printi.h"
int main()
{
  int n = 16, i = 0, j = 1, *seg = &i;

  while (n-- > 0) {
    putchar(' ');
    printi(*seg = i + j, 10);
    seg = (seg == &i) ? &j : &i;
  }
  putchar('\n');
  return 0;
}
