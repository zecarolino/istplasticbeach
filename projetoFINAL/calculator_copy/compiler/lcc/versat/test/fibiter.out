VCD info: dumpfile xtop.vcd opened for output.
 1 2 3 5 8 13 21 34 55 89 144 233 377 610 987 1597
0
Catched Trap at time  51473 clock cycles:
